rust/tw_bitcoin/src/entry.rs

```diff
@@ -0,0 +1,388 @@
+        for output in proto.outputs {
+            let utxo = crate::modules::transactions::OutputBuilder::utxo_from_proto(&output)?;
+
+            utxo_outputs.push(utxo);
+        }
+
+        // Prepare PreSerialization input for Utxo compiler.
+        let utxo_preserializtion = UtxoProto::PreSerialization {
+            version: proto.version,
+            lock_time: proto.lock_time.clone(),
+            inputs: utxo_input_claims.clone(),
+            outputs: utxo_outputs
+                .iter()
+                .map(|out| UtxoProto::TxOut {
+                    value: out.value,
+                    script_pubkey: Cow::Borrowed(&out.script_pubkey),
+                })
+                .collect(),
+            weight_base: proto.fee_per_vb,
+        };
+
+        // Compile the transaction, build the final encoded transaction
+        // containing the signatures/scriptSigs/witnesses.
+        let utxo_serialized = tw_utxo::compiler::Compiler::compile(utxo_preserializtion);
+        handle_utxo_error(&utxo_serialized.error)?;
+
+        // Prepare `Proto::TransactionInput` protobufs for signing output.
+        let mut proto_inputs = vec![];
+        for input in utxo_input_claims {
+            proto_inputs.push(Proto::TransactionInput {
+                txid: Cow::Owned(input.txid.to_vec()),
+                vout: input.vout,
+                sequence: input.sequence,
+                script_sig: Cow::Owned(input.script_sig.into_owned()),
+                witness_items: input
+                    .witness_items
+                    .into_iter()
+                    .map(|item| Cow::Owned(item.into_owned()))
+                    .collect(),
+            });
+        }
+
+        // Prepare `Proto::TransactionOutput` protobufs for output.
+        let mut proto_outputs = vec![];
+        for output in utxo_outputs {
+            proto_outputs.push(Proto::TransactionOutput {
+                script_pubkey: output.script_pubkey,
+                value: output.value,
+                taproot_payload: output.taproot_payload,
+                control_block: output.control_block,
+            });
+        }
+
+        // Prepare `Proto::Transaction` protobuf for output.
+        let transaction = Proto::Transaction {
+            version: proto.version,
+            lock_time: proto.lock_time,
+            inputs: proto_inputs,
+            outputs: proto_outputs,
+        };
+
+        // Return the full protobuf output.
+        Ok(Proto::SigningOutput {
+            error: Proto::Error::OK,
+            transaction: Some(transaction),
+            encoded: utxo_serialized.encoded,
+            txid: utxo_serialized.txid,
+            weight: utxo_serialized.weight,
+            fee: utxo_serialized.fee,
+        })
+    }
+}
+
+// Convenience function for pre-processing of certain fields that must be
+// executed on each `CoinEntry` call.
+fn pre_processor(mut proto: Proto::SigningInput<'_>) -> Proto::SigningInput<'_> {
```

Comment:
I guess so, but this is generally (Bitcoin) fork specific. In a general sense, the Utxo model itself does not mandate this behavior. I guess this could be configurable.

Reply:
Yes but in general it will allow us to avoid code duplicating